### ISSUE TITLE
False positive "should be constant"

### ISSUE THREAD

**Issue by igrnwy** [baldapps/artemis#1]
A function which modifies member variables which are arrays (of POD or classes or STL) is incorrectly flagged as being suitable as a const function. For example, this will claim Test() should be const:
```
class foo
{
    foo();
    ~foo();

    void Test();

    int bar[10];
};

void foo:Test()
{
    bar[3] = 0;
}

```

**Comment by baldapps** [comment]
There was a big in CDT actually fixed in CDT 9.10, eclipse 2019-12. Are you using an older version?

**Comment by igrnwy** [comment]
Ok, yes, that would be it. I have CDT 9.9 and, even if I specify to check the 9.10 URL, it doesn't update. I'm using STM32CubIDE, I guess I'll wait for the vendor to release an update.
Apologies for the flase alarm!
